### DISCUSSION TITLE
Are the mapperly generated files also under license ?

### DISCUSSION THREAD

**Discussion by sidewinder94** [riok/mapperly#781] [Q&A]
As I am evaluating using Mapperly in a corporate project, I was wondering: 

Are the generated mappers covered under the Apache 2.0 license?

After reading the license, my understanding would be:  No.

I would prefer to be sure though.

Will I *need* to include the Apache 2.0 license and copyright notices in the software? (or in a file distributed *with* the software)?

**Comment by latonz** [comment]
IANAL but I think you need to include a copy of the apache 2.0 license since you are using Mapperly in your project. The generated code is probably not subject to copyright.  (see also https://opensource.stackexchange.com/a/9783).

**Reply by sidewinder94** [reply]
I'm not sure, since no binaries and no derivative works from mapperly end up in the published version of the application, only the partial classes generated.

The question would be if some of the code present in those generated files is protected under the Apache 2.0 license.

**Reply by cremor** [reply]
> no binaries and no derivative works from mapperly end up in the published version of the application

How would that be possible? Your code will at least have references to the `MapperAttribute` class from the Riok.Mapperly.Abstractions.dll assembly.

**edit:** Oh, I just found #1027. Didn't know about this before.

**Reply by latonz** [reply]
The references to the attributes get removed by the conditional attribute on the attributes unless you set the conditional symbol (see also [docs](https://mapperly.riok.app/docs/getting-started/installation/#preserving-the-attributes-at-runtime)). The enums of the abstractions don't have the conditional attribute, but as they are usually only used within the attributes which are conditional there shouldn't be any references to the enums left.
As long as you set `ExcludeAssets="runtime"` on the Mapperly package reference only generated Mapperly code is referenced at runtime. However, for some features like reference handling, `ExcludeAssets="runtime"` does not work and Mapperly requires runtime references.
